### DISCUSSION TITLE
How to know which cells were edited?

### DISCUSSION THREAD

**Discussion by danisal** [nick-keller/react-datasheet-grid#96] [Q&A] [answered]
Hi,

Is there a way to know which cells were edited?

If not, any idea on how this could be implemented?

Thanks in advance

**Comment by nick-keller** [comment, marked as answer]
Hello @danisal ,
There are multiple ways to achieve this.
This easiest way is to use the [onBlur](https://react-datasheet-grid.netlify.app/docs/api-reference/props#onblur) prop, only called when the user is done editing a cell. But it will not be called when the user adds a row, deletes a row, copy-pastes data...

The second, mote robust solution is to use `onChange` and to do some with between the received value and the current value. An example implementation of this can be found [here](https://github.com/Equify/react-datasheet-grid/discussions/81#discussioncomment-1674351).

Since this topic is probably the most requested I think it should be part of the core lib.

**Reply by danisal** [reply]
Definitely is something that I would make use of. I'll try to come up with something and make a PR if it would make sense

Cheers

**Reply by nick-keller** [reply]
That'd be great!
Feel free to propose an approach before coding, we might save some time!

**Reply by danisal** [reply]
Probably having some state to keep track of edited/updated cells in conjunction with a CSS class on the cell? I guess it would also be necessary to have a reset function available to the user, in case they just persist the data and want to reset that "updated cells" state...

**Reply by danisal** [reply]
Is there a way to dynamically change a Cell class(name)?

**Reply by nick-keller** [reply]
Yes, [cellClassName](https://react-datasheet-grid.netlify.app/docs/api-reference/columns#cellclassname)

**Comment by nick-keller** [comment]
This is now available in 3.4.0, please read the doc for [`onChange`](https://react-datasheet-grid.netlify.app/docs/api-reference/props#onchange)
